conans/client/graph/graph.py

```diff
@@ -66,39 +44,108 @@ def __init__(self, ref, conanfile, context, recipe=None, path=None):
         self.path = path  # path to the consumer conanfile.xx for consumer, None otherwise
         self._package_id = None
         self.prev = None
-        conanfile._conan_node = self  # Reference to self, to access data
+        if conanfile is not None:
+            conanfile._conan_node = self  # Reference to self, to access data
         self.conanfile = conanfile
-        self.dependencies = []  # Ordered Edges
-        self.dependants = set()  # Edges
+
         self.binary = None
         self.recipe = recipe
         self.remote = None
         self.binary_remote = None
-        self.revision_pinned = False  # The revision has been specified by the user
         self.context = context
 
-        # A subset of the graph that will conflict by package name
-        self._public_deps = _NodeOrderedDict()  # {ref.name: Node}
-        # all the public deps only in the closure of this node
-        # The dependencies that will be part of deps_cpp_info, can't conflict
-        self._public_closure = _NodeOrderedDict()  # {ref.name: Node}
-        # The dependencies of this node that will be propagated to consumers when they depend
-        # on this node. It includes regular (not private and not build requires) dependencies
-        self._transitive_closure = OrderedDict()
-        self.inverse_closure = set()  # set of nodes that have this one in their public
-        self._ancestors = _NodeOrderedDict()  # set{ref.name}
-        self._id = None  # Unique ID (uuid at the moment) of a node in the graph
+        self.id = None  # Unique ID (incremental integer assigned by Graph) of a node in the graph
         self.graph_lock_node = None  # the locking information can be None
-        self.id_direct_prefs = None
-        self.id_indirect_prefs = None
 
-    @property
-    def id(self):
-        return self._id
+        # real graph model
+        self.transitive_deps = OrderedDict()  # of _TransitiveRequirement
+        self.dependencies = []  # Ordered Edges
+        self.dependants = []  # Edges
+        self.error = None
 
-    @id.setter
-    def id(self, id_):
-        self._id = id_
+    def __lt__(self, other):
+        # TODO: Remove this order, shouldn't be necessary
+        return str(self.ref) < str(other.ref)
+
+    def propagate_closing_loop(self, require, prev_node):
+        self.propagate_downstream(require, prev_node)
+        # List to avoid mutating the dict
+        for transitive in list(prev_node.transitive_deps.values()):
+            # TODO: possibly optimize in a bulk propagate
+            prev_node.propagate_downstream(transitive.require, transitive.node, self)
+
+    def propagate_downstream(self, require, node, src_node=None):
+        # print("  Propagating downstream ", self, "<-", require)
```

Comment:
```suggestion
```

Reply:
These have been extraordinary useful for debugging things, I will surely remove them, but please let them stay a bit longer, at least just before merging this.